### PR TITLE
Enhance .gitignore files for better exclusion of unnecessary files

### PR DESCRIPTION
**Pull request by hichemfantar**
Update .gitignore to include additional logs, editor files, and generated artifacts, improving project cleanliness and reducing clutter in version control.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,13 +1,38 @@
+# Logs
+logs
+*.log
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
+pnpm-debug.log*
+lerna-debug.log*
```
Comment (Simek) [inline .gitignore:8]: We should only ignore the files from package manager supported in repo, adding other manager files can obsuscate an user issue.

File: .gitignore [hunk 2]
```diff
@@ -1,13 +1,38 @@
...
+!.vscode/extensions.json
 .idea
-.nvmrc
-.docusaurus
+.DS_Store
+*.suo
+*.ntvs*
+*.njsproj
+*.sln
+*.sw?
```
Comment (Simek) [inline .gitignore:22]: Those entries are totally unrelated to the project.

File: .gitignore [hunk 3]
```diff
@@ -1,13 +1,38 @@
...
-sync-api-docs/generatedComponentApiDocs.js
-sync-api-docs/extracted.json
+# https://yarnpkg.com/getting-started/qa#which-files-should-be-gitignored
+.pnp.*
+.yarn/*
+!.yarn/patches
+!.yarn/plugins
+!.yarn/releases
+!.yarn/sdks
+!.yarn/versions
```
Comment (Simek) [inline .gitignore:33]: This part is for Yarn Berry, it should be added in the PR migrating to it.

File: website/.gitignore [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+# Misc
+.DS_Store
+.env.local
+.env.development.local
+.env.test.local
+.env.production.local
+
+npm-debug.log*
+yarn-debug.log*
+yarn-error.log*
```
Comment (Simek) [inline website/.gitignore:20]: We should stick to one global gitignore at monorepo root, since we don't support sparse checkout anyway.

**Review by Simek — COMMENTED** [review]
In general, I would prefer us to stay with manually managed `.gitignore`, and only add entries in there if any issue arose. 

Would be nice if you can chop down the changeset down to only necessary changes that would be great.

File: .gitignore (hunk 2 above)
Comment (hichemfantar) [reply .gitignore:22]: included these from the vite template but sure i'll get rid of em

Comment (hichemfantar) [reply .gitignore:19]: on second thought, those are actually node config files for visual studio so they should be ignored

Comment (hichemfantar) [reply .gitignore:19]: for reference https://github.com/vitejs/vite/blob/main/packages/create-vite/template-react-ts/_gitignore

File: .gitignore [hunk 5]
```diff
@@ -1,13 +1,31 @@
...
+yarn-error.log*
+
 node_modules
-.DS_Store
+*.local
+
+# Editor directories and files
+.vscode/*
+!.vscode/extensions.json
+!.vscode/settings.json
```
Comment (Simek) [inline .gitignore:13]: Last nit, there is no need to store any VSCode related files since we do not recommend any specific IDE, you can use whatever you want, so let's ignore all editor files:
```suggestion
```
